src/thunderbots/software/ai/hl/stp/play/corner_kick_play.cpp

```diff
@@ -0,0 +1,236 @@
+     *    time decrease the minimum "quality" of pass we'll accept so we're eventually
+     *    forced to at least accept one
+     *  - During this time we continue to run the cherry pick and bait robots
+     * 3. Execute the pass:
+     *  - Once we've decided on a pass, we simply yield a passer/receiver and execute
+     *    the pass
+     *
+     */
+
+    // Figure out if we're taking the kick from the +y or -y corner
+    bool kick_from_pos_corner = world.ball().position().y() > 0;
+
+    // We want the two cherry pickers to be in rectangles on the +y and -y sides of the
+    // field in the +x half. We also further offset the rectangle from the goal line
+    // for the cherry-picker closer to where we're taking the corner kick from
+    Vector pos_y_goalline_x_offset(world.field().enemyDefenseArea().width(), 0);
+    Vector neg_y_goalline_x_offset(world.field().enemyDefenseArea().width(), 0);
+    if (kick_from_pos_corner)
+    {
+        pos_y_goalline_x_offset += {world.field().enemyDefenseArea().width(), 0};
+    }
+    else
+    {
+        // kick from neg corner
+        neg_y_goalline_x_offset += {world.field().enemyDefenseArea().width(), 0};
+    }
+    Vector center_line_x_offset(1, 0);
+    Rectangle pos_y_cherry_pick_rectangle(
+        world.field().centerPoint() + center_line_x_offset,
+        world.field().enemyCornerPos() - pos_y_goalline_x_offset);
+    Rectangle neg_y_cherry_pick_rectangle(
+        world.field().centerPoint() + center_line_x_offset,
+        world.field().enemyCornerNeg() - neg_y_goalline_x_offset);
+
+    // This tactic will move a robot into position to initially take the free-kick
+    auto align_to_ball_tactic = std::make_shared<MoveTactic>();
+
+    // These two tactics will set robots to roam around the field, trying to put
+    // themselves into a good position to receive a pass
+    auto cherry_pick_tactic_pos_y =
+        std::make_shared<CherryPickTactic>(world, pos_y_cherry_pick_rectangle);
+    auto cherry_pick_tactic_neg_y =
+        std::make_shared<CherryPickTactic>(world, neg_y_cherry_pick_rectangle);
+
+    // Setup two bait robots on the opposite side of the field to where the corner kick
+    // is taking place to pull enemies away from the goal
+    Point opposite_corner_to_kick = kick_from_pos_corner ? world.field().enemyCornerNeg()
+                                                         : world.field().enemyCornerPos();
+    Point bait_move_tactic_1_pos =
+        opposite_corner_to_kick - Vector(world.field().enemyDefenseArea().width() * 0.5,
+                                         copysign(0.5, opposite_corner_to_kick.y()));
+    Point bait_move_tactic_2_pos =
+        opposite_corner_to_kick - Vector(world.field().enemyDefenseArea().width() * 1.5,
+                                         copysign(0.5, opposite_corner_to_kick.y()));
+    auto bait_move_tactic_1 = std::make_shared<MoveTactic>(true);
```

Comment:
Were we going to make these patrol instead?

Reply:
We can, not sure what the best patrol strategy would be? Moving over to slack real quick.........